tests/test_metrics.py

```diff
@@ -200,7 +200,7 @@ def test_cuda_available_model_no_device(self, model, is_available_mock):
         is_available_mock.return_value = True
         model.device = None
 
-        with pytest.warns(None):  # assert no warning is raised here
+        with pytest.warns(UserWarning):
```

Comment:
Can you also add the warning match message logic here too so we make sure the expected error is raised and not just any random `UserWarning`?